huxley/core/admin/assignment.py

```diff
@@ -41,29 +42,44 @@ def load(self, request):
         reader = csv.reader(assignments['csv'])
 
         def get_model(model, name, cache):
-            if not name in cache:
-                cache[name] = model.objects.get(name=name)
+            try:
+                if not name in cache:
+                    cache[name] = model.objects.get(name=name)
+            except ObjectDoesNotExist:
+                cache[name] = name
             return cache[name]
 
-        def generate_assigments(reader):
+        def generate_assignments(reader):
             committees = {}
             countries = {}
             schools = {}
+
             for row in reader:
                 if (row[0]=='School' and row[1]=='Committee' and row[2]=='Country'):
                     continue # skip the first row if it is a header
-                committee = get_model(Committee, row[1], committees)
-                country = get_model(Country, row[2], countries)
-                school = get_model(School, row[0], schools)
+
+                while len(row) < 3:
+                    row.append("") # extend the row to have the minimum proper num of columns
+                
                 if len(row) < 4:
                     rejected = False # allow for the rejected field to be null
                 else:
                     rejected = (row[3].lower() == 'true') # use the provided value if admin provides it
-                yield (committee.id, country.id, school.id, rejected)
 
-        Assignment.update_assignments(generate_assigments(reader))
+                committee = get_model(Committee, row[1], committees)
+                country = get_model(Country, row[2], countries)
+                school = get_model(School, row[0], schools)
+                yield (committee, country, school, rejected)
+
+    
+        failed_rows = Assignment.update_assignments(generate_assignments(reader))
+        if failed_rows:
+            messages.error(request, 
+                'Assignment upload aborted. These assignments failed: ' + ' | '.join(failed_rows))
```

Comment:
Maybe we can join with a newline instead of a pipe? Based on the screenshot, it might be overwhelming if there are a lot of errors.